SwiftWisdom/Core/StandardLibrary/Numbers/UnsignedInteger+Extensions.swift

```diff
@@ -26,16 +18,46 @@ public func random(inRange range: CountableClosedRange<Int>) -> Int {
 
 // TODO: write tests for this extension
 extension UnsignedInteger {
-    public init(ip_data: NSData) {
+    public init(ip_data: Data) {
         let hexInt = ip_data.ip_hexInt ?? 0
         self.init(ip_safely: hexInt)
     }
-
-    public init(ip_data: Data) {
+    
+    public init(ip_data: NSData) {
         let hexInt = ip_data.ip_hexInt ?? 0
         self.init(ip_safely: hexInt)
     }
-
+    
+    public init<T : SignedInteger>(ip_safely value: T) {
+        if value < 0 {
+            self.init(ip_safely: UInt8(0))
+        } else {
+            let max = UIntMax(value.toIntMax())
+            self.init(ip_safely: max)
+        }
+    }
+    
+    public init<T : UnsignedInteger>(ip_safely value: T) {
+        self = 0
+        
+        let maxSelf = type(of: self).ip_maxValue
+        if maxSelf.toUIntMax() >= value.toUIntMax() {
+            self = .init(value.toUIntMax())
+        } else {
+            self = maxSelf
+        }
+    }
+    
+    public static func ip_random() -> Self {
+        let intMax = Int(ip_maxValue.toIntMax())
+        let rand = random(inRange: 0...intMax)
+        return self.init(ip_safely: rand)
+    }
+    
+    public func ip_containsMask(_ mask: Self) -> Bool {
```

Comment:
This is confusing. What's the difference between this and `public func ip_containsBitMask(_ bitMask: Self)`?